src/main/scala/com/wavesplatform/network/Coordinator.scala

```diff
@@ -1,162 +1,114 @@
-package scorex.network
+package com.wavesplatform.network
 
-import akka.actor.ActorRef
-import akka.event.LoggingReceive
-import akka.pattern.{ask, pipe}
-import akka.util.Timeout
 import cats._
 import cats.implicits._
-import com.wavesplatform.settings.{BlockchainSettings, WavesSettings}
+import com.wavesplatform.mining.Miner
+import com.wavesplatform.settings.BlockchainSettings
 import com.wavesplatform.state2.ByteStr
 import com.wavesplatform.state2.reader.StateReader
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.local.LocalChannel
+import io.netty.channel.{Channel, ChannelHandlerContext, ChannelInboundHandlerAdapter}
 import scorex.block.Block
+import scorex.block.Block.BlockId
 import scorex.consensus.TransactionsOrdering
-import scorex.consensus.mining.BlockGeneratorController.{LastBlockChanged, StartGeneration}
 import scorex.crypto.EllipticCurveImpl
 import scorex.crypto.encode.Base58
-import scorex.crypto.encode.Base58.encode
-import scorex.network.BlockchainSynchronizer.{GetExtension, GetSyncStatus, Status}
-import scorex.network.NetworkController.{DataFromPeer, SendToNetwork}
-import scorex.network.ScoreObserver.{CurrentScore, GetScore}
-import scorex.network.message.{Message, MessageSpec, _}
-import scorex.network.peer.PeerManager.{ConnectedPeers, GetConnectedPeersTyped}
-import scorex.transaction.History.BlockchainScore
 import scorex.transaction.ValidationError.GenericError
 import scorex.transaction._
 import scorex.utils.{ScorexLogging, Time}
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
-import scala.language.{higherKinds, postfixOps}
 import scala.util.control.NonFatal
-import com.wavesplatform.state2._
-
-class Coordinator(protected override val networkControllerRef: ActorRef, blockchainSynchronizer: ActorRef, blockGenerator: ActorRef,
-                  peerManager: ActorRef, scoreObserver: ActorRef, blockchainUpdater: BlockchainUpdater, time: Time,
-                  utxStorage: UnconfirmedTransactionsStorage,
-                  history: History, stateReader: StateReader, checkpoints: CheckpointService, settings: WavesSettings) extends ViewSynchronizer with ScorexLogging {
 
+@Sharable
+class Coordinator(
```

Comment:
We might argue whether `ExtensionLoader` is a protocol detail or not, but Coordinator for sure isn't. Its business logic needs to be clear from any `netty` implementation details